modules/xcode/tests/test_xcode4_project.lua

```diff
@@ -18,11 +18,25 @@
 
 	xcode.used_ids = {}
 
+--
+-- Historically, only the first two items passed into xcode.newid were printed for testing
+--
+	xcode.id_print_depth = 2
+
 	local old_idfn = xcode.newid
-	xcode.newid = function(node, usage)
-		local name = node
-		if usage then
-			name = name .. ":" .. usage
+	xcode.newid = function(...)
```

Comment:
I am not opposed to removing the override, however I believe keeping it for testing purposes has value. The comment above the override describes the reasons for doing so.

Unfortunately, the testing override was not updated with when the original xcode.newid [changed](https://github.com/premake/premake-xcode/commit/210e74fafe9d837965e037f2abfadc6f1d7a2e00#diff-ef36fc761013ebfa60a3b283be4294d1L246) from taking two parameters to a variable number of them.

Would you be opposed to keeping the testing override and changing it to always print all the inputs used for hashcode calculation (instead of just two)? If you would rather just have the calculated hashcode in the testing output, just say the word!

Reply:
Given that we build for Linux, macOS and Windows and run the tests on them every time a PR is opened, wouldn't that be enough to catch the non-portable paths being used? At the very least we shouldn't randomly limit the count, this would be far too easy for people to just ignore the path parameter because it was different across platforms or because they didn't know that `id_print_depth` existed.

Reply:
> Given that we build for Linux, macOS and Windows and run the tests on them every time a PR is opened, wouldn't that be enough to catch the non-portable paths being used?

Hmmm, yeah I see your point. I think I'm good with deleting the override and updating the tests as necessary (there are a lot of tests)

Reply:
Sounds good 😄

Reply:
@samsinsane rebased, squashed and tested. Should be good to go!